src/Corax/Utils/EntryIdEncodings.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+namespace Corax.Utils;
+
+public static class EntryIdEncodings
+{
+    //Schema:
+    //[CONTAINER_ID][FREQUENCY][CONTAINER_TYPE]
+    
+    // Positions:
+    private const int FrequencySizeInBits = 8;
+    private const int EntryIdOffset = FrequencySizeInBits + ContainerTypeOffset;
+    private const long Mask = 0xFFL;
+    private const int ContainerTypeOffset = 2;
+
+    
+    
+    private const int AllBuffer = -1;
+    
+    // Quantization parameters:
+    private const long Min = 0;
+    private const long Max = 255;
+    private const long QuantizationMax = 256 - 1;
+    // ReSharper disable once PossibleLossOfFraction
+    private const double QuantizationStep = (Max - Min) / (double)QuantizationMax;
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static long Encode(long entryId, short count, TermIdMask containerType)
+    {
+        Debug.Assert((count & ~Mask) == 0);
```

Comment:
Need to check here that `entryId` is smaller than `2^54` here

Reply:
I would introduce a new constant here and create the mask based on that constant and explain what is the rationale for the constant construction in place. The check will never happen so it is 100% predicted (aka cheap as f&%$*)